### PR TITLE
Fixes to operator CSV/catalog and example CR

### PR DESCRIPTION
**Pull request by avalluri**
CSV:
- add missing CRD 'displayName','description' fields.
- Deployment -> PmemCSIDeployment naming changes
- mention NFD usage
example CR:
 - set `storage: pmem` as the default node selector.

### CODE REVIEW

File: deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ spec:
   deviceMode: "lvm"
   nodeSelector:
     # When using Node Feature Discovery (NFD):
-    feature.node.kubernetes.io/memory-nv.dax: "true"
+    # feature.node.kubernetes.io/memory-nv.dax: "true"
     # When using manual node labeling with that label:
-    # storage: pmem
+    storage: pmem
```
Comment (pohly) [inline deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml:11]: Why this change?

Comment (pohly) [reply deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml:11]: We shouldn't encourage users to use "storage" as label. It's a bad choice for a label because it isn't namespaced and may conflict with other usage of labels.

We are kind of stuck with it because we promised seamless updates from previous versions, but for someone who is starting out anew (and people using the operator will be in that camp) it makes sense to pick a better label.

The NFD label can also be set manually in case that NFD is not used.

Comment (avalluri) [reply deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml:11]: > Why this change?

There is a mismatch between the documentation and the example CR, an the example is not usable out of the box in a cluster where no NFD installed. 

> but for someone who is starting out anew (and people using the operator will be in that camp) it makes sense to pick a better label.
> 
> The NFD label can also be set manually in case that NFD is not used.

Then shall we replace the node label name in the OperatorHub documentation as below:

```
**Label the cluster nodes that provide persistent memory device(s)**

$ kubectl label node <your node> feature.node.kubernetes.io/memory-nv.dax="true"

You can skip this step if your cluster is installed with [Node Feature Discovery(NFD)](https://kubernetes-sigs.github.io/node-feature-discovery/stable/get-started/index.html)

```

Comment (pohly) [reply deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml:11]: Which documentation is "the documentation"? https://intel.github.io/pmem-csi/latest/docs/install.html#installation-and-setup ?

It introduces the "storage=pmem" label as a "historic" choice. That some other label like the one from NFD is less likely to conflict is not called out explicitly, but could be added.

> Then shall we replace the node label name in the OperatorHub documentation as below:
[...]

Yes, I think that is better.

But how much documentation do we really want to store on OperatorHub? Do we expect that people will be able to use the operator solely based on information there? I'm worried that we'll end up maintaining our documentation in two places.

Comment (pohly) [reply deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml:11]: "You can skip this step if your cluster is installed with" -> "You can skip this step if your cluster has NFD installed"

Comment (avalluri) [reply deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml]: > But how much documentation do we really want to store on OperatorHub?

Very minimal. It just has software/hardware requirements and nodes setup as part of pre-requisites.

Comment (avalluri) [reply deploy/common/pmem-csi.intel.com_v1beta1_pmemcsideployment_cr.yaml]: > "You can skip this step if your cluster is installed with" -> "You can skip this step if your cluster has NFD installed"

Done.

File: deploy/kustomize/olm-catalog/bases/pmem-csi-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -109,23 +111,27 @@ spec:
...
     ```
     **Label the cluster nodes that provide persistent memory device(s)**
+
+    You can skip this step if your cluster has [Node Feature Discovery((NFD)](https://kubernetes-sigs.github.io/node-feature-discovery/stable/get-started/index.html) installed.
     ```
-    $ kubectl label node <your node> storage=pmem
+    $ kubectl label node <your node> feature.node.kubernetes.io/memory-nv.dax="true"
     ```
-    
+
```
Comment (pohly) [inline deploy/kustomize/olm-catalog/bases/pmem-csi-operator.clusterserviceversion.yaml:119]: We discussed that the setup instructions on operator hub should not repeat content from our main documentation site and instead guide users toward that.

Do you intend to implement that in this PR or separately?

Comment (avalluri) [reply deploy/kustomize/olm-catalog/bases/pmem-csi-operator.clusterserviceversion.yaml:119]: > Do you intend to implement that in this PR or separately?

I will prepare a separate PR.

Comment (pohly) [reply deploy/kustomize/olm-catalog/bases/pmem-csi-operator.clusterserviceversion.yaml:119]: I'm not sure why we need to merge this PR, but okay.

I opened an issue so that we don't forget: https://github.com/intel/pmem-csi/issues/891
